btp-document/Smart-Contracts-PRA.md

```diff
@@ -0,0 +1,530 @@
+   > account:             0xf24FF3a9CF04c71Dbc94D0b566f7A27B94566cac
+   > balance:             1207825.601491923174706176
+   > gas used:            4535311 (0x45340f)
+   > gas price:           20 gwei
+   > value sent:          0 ETH
+   > total cost:          0.09070622 ETH
+
+
+   Deploying 'ProxyAdmin'
+   ----------------------
+   > transaction hash:    0xe515f6aff766cec9a3af1ffd0cafd8c6af9763d5d1633d630a9d7df0ad7555d4
+   > Blocks: 0            Seconds: 0
+   > contract address:    0x294c664f6D63bd1521231a2EeFC26d805ce00a08
+   > block number:        18
+   > block timestamp:     1626151633
+   > account:             0xf24FF3a9CF04c71Dbc94D0b566f7A27B94566cac
+   > balance:             1207825.591837523174706176
+   > gas used:            482720 (0x75da0)
+   > gas price:           20 gwei
+   > value sent:          0 ETH
+   > total cost:          0.0096544 ETH
+
+
+   Deploying 'TransparentUpgradeableProxy'
+   ---------------------------------------
+   > transaction hash:    0x9d4e422af278e00133dfe05b04626035fe1eb5e24972c026fa78ff42ddbcf38b
+   > Blocks: 0            Seconds: 0
+   > contract address:    0x82745827D0B8972eC0583B3100eCb30b81Db0072
+   > block number:        19
+   > block timestamp:     1626151634
+   > account:             0xf24FF3a9CF04c71Dbc94D0b566f7A27B94566cac
+   > balance:             1207825.574454443174706176
+   > gas used:            869154 (0xd4322)
+   > gas price:           20 gwei
+   > value sent:          0 ETH
+   > total cost:          0.01738308 ETH
+
+
+   Replacing 'BSHPeriphery'
+   ------------------------
+   > transaction hash:    0x50b4942f25173db2a9d7a17a749f4266e18c254c52a84d4e82f98b2bef2b0c3d
+   > Blocks: 0            Seconds: 0
+   > contract address:    0xEC69d4f48f4f1740976968FAb9828d645Ad1d77f
+   > block number:        20
+   > block timestamp:     1626151636
+   > account:             0xf24FF3a9CF04c71Dbc94D0b566f7A27B94566cac
+   > balance:             1207825.512091723174706176
+   > gas used:            3118136 (0x2f9438)
+   > gas price:           20 gwei
+   > value sent:          0 ETH
+   > total cost:          0.06236272 ETH
+
+
+   Deploying 'TransparentUpgradeableProxy'
+   ---------------------------------------
+   > transaction hash:    0x98ee35989b5b21ca4fc222e7bc639b40cc25639f6029ff7eb2f41354985b9e47
+   > Blocks: 0            Seconds: 0
+   > contract address:    0x493275370aF3f63d9ccd10a6539435121cF4fbb9
+   > block number:        21
+   > block timestamp:     1626151636
+   > account:             0xf24FF3a9CF04c71Dbc94D0b566f7A27B94566cac
+   > balance:             1207825.497248583174706176
+   > gas used:            742157 (0xb530d)
+   > gas price:           20 gwei
+   > value sent:          0 ETH
+   > total cost:          0.01484314 ETH
+
+   > Saving artifacts
+   -------------------------------------
+   > Total cost:          0.19494956 ETH
+
+
+Summary
+=======
+> Total deployments:   5
+> Final cost:          0.19494956 ETH
+```
```

Comment:
Need to tell the user to save the deployed contract address because it's used later